src/sage/categories/finite_fields.py

```diff
@@ -243,5 +246,222 @@ def _element_of_factored_order(self, F):
                     return a
             raise AssertionError("no element found")
 
+        @cached_method
+        def quadratic_non_residue(self):
+            """
+            Returns a random non square element of the finite field
+
+            OUTPUTS:
+                - a non-square element of the finite field; raises an error if
+                  the finite field is of even order
+
+            EXAMPLES::
+                sage: k = GF((3, 10))
+                sage: k.quadratic_non_residue().is_square()
+                False
+                sage: k = GF((2, 10))
+                sage: k in Fields()  # to let k be a finite field
+                True
+                sage: k.quadratic_non_residue()
+                Traceback (most recent call last):
+                ...
+                ValueError: There are no non-squares in finite fields of even order
+            """
+            # if the order is an even power of two
+            # then every element is a square
+            if self.order() % 2 == 0:
+                raise ValueError("There are no non-squares in finite fields of even order")
+            # uniformly randomly select elements for a non-square
+            # with probability 1/2 for a non-square
+            element = self.random_element()
+            while element.is_square():
+                element = self.random_element()
+            return element
+
     class ElementMethods:
-        pass
+        def is_square(self) -> bool:
+            """
+            Tests if the element is a square or has
+            a square root element
+
+            OUTPUT:
+            - `True` if the element is a square `False` if not
+
+            EXAMPLES::
```

Comment:
```suggestion
            EXAMPLES::

```

There needs to be a blank line after EXAMPLES